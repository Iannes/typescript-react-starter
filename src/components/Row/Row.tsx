import React from "react";
import { RowProps } from "types";
import "./Row.css";

export const Row: React.FC<RowProps> = props => {
  const {
    rowKeyName,
    rowAlternativeTitle = "",
    hideTitle = false,
    children
  } = props;

  return (
    <section className="row">
      {!hideTitle ? <span>{rowAlternativeTitle || rowKeyName}</span> : null}
      {children}
    </section>
  );
};
