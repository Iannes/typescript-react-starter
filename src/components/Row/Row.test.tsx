import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import Row from "./";

beforeEach(() => {
  jest.resetModules();
});

describe("<Row />", () => {
  test("Renders properly", () => {
    const component = renderer.create(<Row rowKeyName={"rowKeyName"} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("If Row doesn't receive hideTitle and rowAlternativeTitle props it shows the keyName for title", () => {
    const wrapper = shallow(<Row rowKeyName={"rowKeyName"} />);
    const row = wrapper.find(".row > span");
    expect(row.text().includes("rowKeyName")).toBe(true);
  });
  test("If Row doesn't receive hideTitle but gets rowAlternativeTitle it shows rowAlternativeTitle for title", () => {
    const wrapper = shallow(
      <Row
        rowAlternativeTitle={"rowAlternativeTitle"}
        rowKeyName={"rowKeyName"}
      />
    );
    const row = wrapper.find(".row > span");
    expect(row.text().includes("rowAlternativeTitle")).toBe(true);
  });
  test("If Row receives hideTitle there is no title", () => {
    const wrapper = shallow(
      <Row
        rowAlternativeTitle={"rowAlternativeTitle"}
        rowKeyName={"rowKeyName"}
        hideTitle
      />
    );
    expect(wrapper.find(".row").children()).toHaveLength(0);
  });
});
