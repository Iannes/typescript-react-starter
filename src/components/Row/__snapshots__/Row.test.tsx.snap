// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Row /> Renders properly 1`] = `
<section
  className="row"
>
  <span>
    rowKeyName
  </span>
</section>
`;
